source/geod24/LocalRest.d

```diff
@@ -770,10 +770,13 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
 
                     auto sel_ret = C.select(read_list, write_list, 10.msecs);
 
-                    if (!control.isSleeping())
+                    if (await_msgs.length > 0 && !control.isSleeping())
                     {
-                        foreach (ref await_msg; await_msgs)
-                            handleCommand(await_msg.var.cmd, node, control.filter, await_msg.conn.resp_chn);
+                        auto await_msgs_duped = await_msgs.dup;
+                        scheduler.spawn({
+                            foreach (ref await_msg; await_msgs_duped)
+                                handleCommand(await_msg.var.cmd, node, control.filter, await_msg.conn.resp_chn);
+                        });
```

Comment:
This defeats the purpose of re-using a buffer. Is there no other way this could be handled ?

Reply:
I dont think there is, without some overkill data structures. But I think we are better off reallocating after each sleep rather than duplicating and reusing the buffer.

Reply:
Makes sense. Let's go with this.